ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/EnumEntryNameCaseRule.kt

```diff
@@ -0,0 +1,59 @@
+package com.pinterest.ktlint.ruleset.experimental
+
+import com.pinterest.ktlint.core.Rule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
+import org.jetbrains.kotlin.psi.KtEnumEntry
+
+class EnumEntryNameCaseRule : Rule("enum-entry-name-case") {
+
+    companion object {
+        const val ERROR_MESSAGE = "Enum entry name should be uppercase underscore-separated names like \"ENUM_ENTRY\" or upper camel-case like \"EnumEntry\""
+    }
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        if (node !is CompositeElement) {
+            return
+        }
+        val enumEntry = node.psi as? KtEnumEntry ?: return
+        val name = enumEntry.name ?: return
+
+        if (name.containsLowerCase()) {
+            // In case of lower camel case like "enumName", or all lower case like "enumname"
+            if (!name.startsWithUpperCase()) {
+                emit(
+                    node.startOffset,
+                    ERROR_MESSAGE, false
+                )
+
+                if (autoCorrect) correct(enumEntry, name)
```

Comment:
This will make breaking format on code change. Maybe you forgot to push disabling formatting?

Reply:
`canBeAutoCorrected` is false, so `autoCorrect` never be `true`.

Reply:
Sorry, missed that :+1: